storage/objects.go

```diff
@@ -304,3 +321,48 @@ func (s *ObjectsClient) Put(ctx context.Context, input *PutObjectInput) error {
 
 	return nil
 }
+
+func createDirectory(c ObjectsClient, ctx context.Context, fullPath string) error {
+	log.Printf("Forcing the Creation of the Directory Tree: %q", fullPath)
+	dirClient := &DirectoryClient{
+		client: c.client,
+	}
+
+	parts := strings.Split(fullPath, "/")
+	parts = parts[:len(parts)-1]
+
+	var folderPath string
+	for i, part := range parts[1:] {
+		if i == 0 {
+			folderPath = part
+			continue
+		}
+		folderPath = path.Clean(path.Join(folderPath, part))
+		err := dirClient.Put(ctx, &PutDirectoryInput{
+			DirectoryName: folderPath,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkDirectoryTreeExists(c ObjectsClient, ctx context.Context, fullPath string) error {
+	dirClient := &DirectoryClient{
+		client: c.client,
+	}
+
+	folders, _ := path.Split(fullPath)
+
+	_, err := dirClient.List(ctx, &ListDirectoryInput{
```

Comment:
Any chance of using our new HEAD functions, `Objects.GetInfo` or `Objects.IsDir` here?

Reply:
This is a check to see if the folder path exists - not if it's a directory or an object

Reply:
Changed to using IsDir :)